Use the right C++ standard library when linking.
This patch needs to be sent to the developers.
--- x2goclient.pro.orig	2020-02-13 04:36:07.000000000 -0600
+++ x2goclient.pro	2020-03-20 05:13:17.000000000 -0500
@@ -236,6 +236,7 @@
 
   !isEmpty(OSX_STDLIB) {
     QMAKE_CXXFLAGS += -stdlib=$${OSX_STDLIB}
+    QMAKE_LFLAGS += -stdlib=$${OSX_STDLIB}
   }
 
   !isEmpty(MACPORTS_INCLUDE_PATH) {
